tempodb/backend/raw.go

```diff
@@ -222,25 +241,18 @@ func (r *reader) BlockMeta(ctx context.Context, blockID uuid.UUID, tenantID stri
 
 // TenantIndex implements backend.Reader
 func (r *reader) TenantIndex(ctx context.Context, tenantID string) (*TenantIndex, error) {
-	reader, size, err := r.r.Read(ctx, TenantIndexName, KeyPath([]string{tenantID}), nil)
+	tenantIndexProto, err := r.tenantIndexProto(ctx, tenantID)
 	if err != nil {
 		return nil, err
 	}
 
-	defer reader.Close()
-
-	bytes, err := tempo_io.ReadAllWithEstimate(reader, size)
+	tenantIndex := &TenantIndex{}
```

Comment:
if the proto read fails should we fall back to reading the old json index? this might help with a smooth transition when all components first come up and there is no proto indexes

Reply:
I thought about this and decided not, though let me know if you disagree.  My thought was that if we are not updating the old file, then it could potentially hang out for years and become completely wrong, so falling back doesn't help.  If we wanted to write both for a period, we could fallback, but I'm not sure its worth the extra complexity.  Presumably we would do this as a hardcode which would be on a release cycle, or as a config flag, which requires extra operational handling.

Reply:
> then it could potentially hang out for years and become completely wrong, so falling back doesn't help.

yeah, good call out. i prefer your approach. we'll just need a bit of extra care on initial rollout, but i know that's on your list regardless